maltoolbox/attackgraph/attackgraph.py

```diff
@@ -493,6 +333,176 @@ def attach_attackers(self) -> None:
 
             attacker.entry_points = list(attacker.reached_attack_steps)
 
+    def _follow_expr_chain(
+            self,
+            model: Model,
+            target_assets: list[Any],
+            expr_chain: ExpressionsChain
+        ) -> list:
+        """
+        Recursively follow a language graph expressions chain on an instance
+        model.
+
+        Arguments:
+        model           - a maltoolbox.model.Model on which to follow the
+                          expressions chain
+        target_assets   - the list of assets that this expressions chain
+                          should apply to. Initially it will contain the
+                          asset to which the attack step belongs
+        expr_chain      - the expressions chain we are following
+
+        Return:
+        A list of all of the target assets.
+        """
+
+        if expr_chain is None:
+            # This is an attack step targeting the same assets
```

Comment:
I think it would be nice to have an example of how this would look like in mal code in each of the cases. For instance, here, it is hard for me to visualize what you are saying in the code. I am sure it is simple, but i just can't get my head around it.

Reply:
I get it now. You pass all children of an attack step through this method here and this conditional applies to a simple step that is reachable, ie:

```
| a_step
  -> another_step
```

Reply:
Perhaps:

```suggestion
            # This is an attack step in the same asset as the parent:
            # | parent_step
            #   -> this_step
```